sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/internal/state/SdkObservableMeasurementTest.java

```diff
@@ -88,12 +85,7 @@ void recordLong_ImmutableData() {
     try {
       sdkObservableMeasurement.record(5);
 
-      verify(mockAsyncStorage1).record(measurementArgumentCaptor.capture());
-      Measurement passedMeasurement = measurementArgumentCaptor.getValue();
-      assertThat(passedMeasurement).isInstanceOf(ImmutableMeasurement.class);
-      assertThat(passedMeasurement.longValue()).isEqualTo(5);
-      assertThat(passedMeasurement.startEpochNanos()).isEqualTo(0);
-      assertThat(passedMeasurement.epochNanos()).isEqualTo(10);
+      verify(mockAsyncStorage1).record(Attributes.empty(), 5);
```

Comment:
Let's retain the asserts to verify the epoch informatinon was set. Applies to other tests as well. 
```suggestion
      verify(mockAsyncStorage1).setEpochInformation(0, 10);
      verify(mockAsyncStorage1).record(Attributes.empty(), 5);
```

Reply:
Point taken, I'll fix this after we clarify the interface in [this thread](https://github.com/open-telemetry/opentelemetry-java/pull/7264#discussion_r2043056229)

Reply:
I added a dedicated test: `SdkObservableMeasurementTest#setActiveReader_SetsEpochInformation`: 4b3f8741f739b8ce6ad8db58d3b7add176acaaa0